File: ant-sdk/antd-elixir/examples/01_connect.exs
Mix.install([
  {:antd, path: ".."}
])

# Connect to the antd daemon and check health
client = Antd.Client.new()

case Antd.Client.health(client) do
  {:ok, health} ->
    IO.puts("Daemon healthy: #{health.ok}")
    IO.puts("Network: #{health.network}")

  {:error, error} ->
    IO.puts("Failed to connect: #{Exception.message(error)}")
    IO.puts("Make sure antd is running: ant dev start")
end

File: ant-sdk/antd-lua/examples/01-connect.lua
--- Example: Connect to antd daemon and check health.

local antd = require("antd")

-- Create a client with default settings (localhost:8082)
local client = antd.new_client()

-- Check daemon health
local health, err = client:health()
if err then
    print("Error: " .. err.message)
    os.exit(1)
end

print("Daemon healthy: " .. tostring(health.ok))
print("Network: " .. health.network)
